source/extensions/quic_listeners/quiche/envoy_quic_dispatcher.cc

```diff
@@ -25,6 +25,13 @@ EnvoyQuicDispatcher::EnvoyQuicDispatcher(
   // Network::UdpListenerCallbacks which should be called at the beginning
   // of HandleReadEvent(). And this callback should call quic::Dispatcher::ProcessBufferedChlos().
   SetQuicFlag(FLAGS_quic_allow_chlo_buffering, false);
+  // Set send buffer twice of max flow control window to ensure that stream send
+  // buffer always takes all the data.
+  // The max amount of data buffered is the per-stream high watermark + the max
+  // flow control window of upstream. The per-stream high watermark should be
+  // smaller than max flow control window to make sure upper stream can be flow
+  // control blocked early enough not to send more than the threshold allows.
+  SetQuicFlag(FLAGS_quic_buffered_data_threshold, 2 * 16 * 1024 * 1024); // 32MB
```

Comment:
This seems pretty huge compared to what we typically recommend for HTTP/2 at the edge. Will we eventually make this configurable?

Reply:
This threshold needs to be large enough to hold all the data from upstream even after readDisable(). So it should be larger than HTTP2 per stream flow control window whose negotiated value is not accessible at this level. I will add a TODO to plumb through to get that value. 

But before that, for QUICHE to work correctly, we need this threshold to be large enough what each encodeData() can consume all the data passed in. This doesn't mean that we are actually buffering that much data because upstream's flow control window probably is not that large after negotiation.